pubsub/rabbitmq/metadata.go

```diff
@@ -14,6 +14,7 @@ type metadata struct {
 	deleteWhenUnused bool
 	autoAck          bool
 	requeueInFailure bool
+	backOffEnable    bool  // Enable backoff or not
```

Comment:
Would it make more sense to have backOff enabled by default and disabled by setting `backOffMaxRetries = 0`?  I'm thinking this setting may not be necessary. Thoughts?

Reply:
Yes, that's makes sense. I will fix it.

A small additional points: the `hazelcast` and `mqtt` component's `backOffMaxRetries` option is defaults to `"0"` which means the component will not retry processing the message. And for this component its default value is `-1` which means retry indefinitely. If there is any necessary to keep it consistent across all components.

Reply:
Yeah, that's a great point. Here is my thought:

For brokers that retry at a per-message level, the default can be `0` and dapr-level retries are "opt in" (because the broker handles retries on its own).
For brokers that are offset/position based (like Kafka), the indefinite retries are essential or messages are not potentially dropped/not processed. Once pubsub has something like dead letter/retry queues, a `0` default value would be "safe".

Reply:
Added another default config `DefaultConfigWithNoRetry` in internal/retry, which set the value of `MaxRetries` to 0. For brokers that retry at a per-message level, can use this config as its default config.